### PR TITLE
move funcs from test/e2e/framework to test/e2e/apps

### PR DESCRIPTION
**Pull request by tanjunchen**
**What type of PR is this?**
 
/kind cleanup

**What this PR does / why we need it**:

move funcs from test/e2e/framework to test/e2e/apps

**Which issue(s) this PR fixes**:

Ref #86052

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

### CODE REVIEW

File: test/e2e/apps/framework.go [hunk 1]
```diff
@@ -16,9 +16,51 @@ limitations under the License.
...
+	"k8s.io/kubernetes/test/e2e/framework"
+)
 
 // SIGDescribe annotates the test with the SIG label.
 func SIGDescribe(text string, body func()) bool {
 	return ginkgo.Describe("[sig-apps] "+text, body)
 }
+
+// WaitForReplicaSetDesiredReplicas waits until the replicaset has desired number of replicas.
+func WaitForReplicaSetDesiredReplicas(rsClient appsclient.ReplicaSetsGetter, replicaSet *appsv1.ReplicaSet) error {
```
Comment (SataQiu) [inline test/e2e/apps/framework.go]: Thanks @tanjunchen 
Would you consider creating a new go file (such as `test/e2e/apps/wait.go`) to hold these code ?

File: test/e2e/apps/framework.go [hunk 2]
```diff
@@ -16,9 +16,51 @@ limitations under the License.
...
+		return rs.Status.ObservedGeneration >= desiredGeneration && rs.Status.Replicas == *(replicaSet.Spec.Replicas) && rs.Status.Replicas == *(rs.Spec.Replicas), nil
+	})
+	if err == wait.ErrWaitTimeout {
+		err = fmt.Errorf("replicaset %q never had desired number of replicas", replicaSet.Name)
+	}
+	return err
+}
+
+// WaitForReplicaSetTargetSpecReplicas waits for .spec.replicas of a RS to equal targetReplicaNum
+func WaitForReplicaSetTargetSpecReplicas(c clientset.Interface, replicaSet *appsv1.ReplicaSet, targetReplicaNum int32) error {
```
Comment (SataQiu) [inline test/e2e/apps/framework.go]: Do these functions have to be exported？ If not, maybe we can keep it private.

File: test/e2e/apps/framework.go (hunk 1 above)
Comment (tanjunchen) [reply test/e2e/apps/framework.go]: Sure. i will do it

File: test/e2e/apps/framework.go (hunk 2 above)
Comment (tanjunchen) [reply test/e2e/apps/framework.go]: Thanks your advice.

File: test/e2e/apps/wait.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
+/*
```
Comment (oomichi) [inline test/e2e/apps/wait.go]: Why don't we move functions into ` test/e2e/apps/deployment.go` directly?
These functions are used in ` test/e2e/apps/deployment.go` only.

Comment (oomichi) [reply test/e2e/apps/wait.go]: Ah I am asking @SataQiu  at the above

Comment (SataQiu) [reply test/e2e/apps/wait.go]: I think it's okay to move functions into `test/e2e/apps/deployment.go` @oomichi 
Just don't recommend moving these functions into `test/e2e/apps/framework.go`.
Can you update it again? @tanjunchen

Comment (tanjunchen) [reply test/e2e/apps/wait.go]: Sure @oomichi @SataQiu Thanks advice

**Pull request merged**